core/src/main/scala/kafka/server/DynamicConfigs.scala

```diff
@@ -0,0 +1,89 @@
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import java.util.Properties
+
+import org.apache.kafka.common.config.ConfigDef
+import org.apache.kafka.common.config.ConfigDef.Importance._
+import org.apache.kafka.common.config.ConfigDef.Type._
+
+/**
+  * Class used to hold dynamic configs. These are configs which have no physical manifestation in the server.properties
+  * and can only be set dynamically.
+  */
+object DynamicConfigs {
+
+  object Broker {
+    //Properties
+    val ThrottledReplicationRateLimitProp = "replication.quota.throttled.rate"
+
+    //Defaults
+    val DefaultThrottledReplicationRateLimit = ReplicationQuotaManagerConfig.QuotaBytesPerSecondDefault
+
+    //Documentation
+    val ThrottledReplicationRateLimitDoc = "A long representing the upper bound (bytes/sec) on replication traffic for replicas enumerated in the " +
+      s"property $ThrottledReplicationRateLimitProp. This property can be only set dynamically via the config command etc."
+
+    //Definitions
+    val brokerConfigs = new ConfigDef()
+      .define(ThrottledReplicationRateLimitProp, LONG, DefaultThrottledReplicationRateLimit, MEDIUM, ThrottledReplicationRateLimitDoc)
+
+    def names() = {
+      brokerConfigs.names()
+    }
+
+    def validateNames(props: Properties): Unit = {
+      DynamicConfigs.validateNames(brokerConfigs, props)
+    }
+  }
+
+  object Client {
+    //Properties
+    val ProducerOverrideProp = "producer_byte_rate"
+    val ConsumerOverrideProp = "consumer_byte_rate"
+
+    //Defaults
+    val DefaultProducerOverride = ClientQuotaManagerConfig.QuotaBytesPerSecondDefault
+    val DefaultConsumerOverride = ClientQuotaManagerConfig.QuotaBytesPerSecondDefault
+
+    //Documentation
+    val ProducerOverrideDoc = "A rate representing the upper bound (bytes/sec) for producer traffic."
+    val ConsumerOverrideDoc = "A rate representing the upper bound (bytes/sec) for consumer traffic."
+
+    //Definitions
+    val clientConfigs = new ConfigDef()
+      .define(ProducerOverrideProp, LONG, DefaultProducerOverride, MEDIUM, ProducerOverrideDoc)
+      .define(ConsumerOverrideProp, LONG, DefaultConsumerOverride, MEDIUM, ConsumerOverrideDoc)
+
+    def names() = {
+      clientConfigs.names()
+    }
+
+    def validateNames(props: Properties): Unit = {
+      DynamicConfigs.validateNames(clientConfigs, props)
+    }
+  }
+
+  private def validateNames(configDef: ConfigDef, props: Properties) {
+    import scala.collection.JavaConversions._
```

Comment:
We generally favour JavaConverters now. You can then have the import at the top level and use `asJava` or `asScala` whenever a conversion is needed.